io/eolearn/tests/test_sentinelhub_process.py

```diff
@@ -164,6 +167,47 @@ def test_upsampling_downsampling(self, resampling_type: ResamplingType, stats: L
         width, height = self.size
         assert bands.shape == (4, height, width, 1)
 
+    @pytest.mark.parametrize(
+        ["geometry", "stats"],
+        [
+            (
+                Geometry(
+                    BBox([easting + 50 * 10, northing, easting + size[0] * 10, northing + 50 * 10], crs=crs).geometry,
+                    crs=crs,
+                ),
+                [0.0, 0.1547, 0.0],
+            ),
+            (
+                Geometry(
+                    BBox(
+                        [easting - 20, northing - 20, easting + (size[0] + 10) * 10, northing + (size[1] + 10) * 10],
+                        crs=crs,
+                    ).geometry,
+                    crs=crs,
+                ),
+                [0.0836, 0.1547, 0.0794],
+            ),
+        ],
+    )
+    def test_geometry_argument(self, geometry: Geometry, stats: List[float]):
+        task = SentinelHubInputTask(
+            bands_feature=(FeatureType.DATA, "BANDS"),
+            bands=["B01"],
+            size=self.size,
+            maxcc=self.maxcc,
+            time_difference=self.time_difference,
+            data_collection=DataCollection.SENTINEL2_L1C,
+            max_threads=self.max_threads,
+        )
+
+        eopatch = task.execute(bbox=self.bbox, time_interval=self.time_interval, geometry=geometry)
+        bands = eopatch[(FeatureType.DATA, "BANDS")]
+
+        assert calculate_stats(bands) == approx(stats)
+
+        width, height = self.size
+        assert bands.shape == (4, height, width, 1)
```

Comment:
Is there a reason why "all" bands are $(4,x,y,1)$ ? or $4$ and $1$ shouldn't be "hard coded"

Reply:
I guess given the bbox and time interval we know there will be 4 timestamps, while given which bands we require in the test we set the last dimension, i.e. number of channels

Surely these tests could be beautified and simplified